src/main/java/org/reactome/addlinks/AddLinks.java

```diff
@@ -193,96 +197,86 @@ public void doAddLinks() throws Exception
-		Files.write(Paths.get(diffReportName), linksReport.getBytes() );
-
 		logger.info("Process complete.");
 	}
 
 	/**
-	 * Checks the links to external resources. Returns a
-	 * @return
+	 * @param linkCheckReportLines
+	 * @param linkCheckManager
+	 * @param refDBInst
+	 * @param reportLine
+	 * @param numLinkOK
+	 * @param numLinkNotOK
 	 */
-	private String checkLinks()
-	{
-		StringBuilder linkCheckReportLines = new StringBuilder("RefDBName\tNumOK\tNumNotOK\n");
-		// Now, check the links that were created to ensure that they are all valid.
-		LinkCheckManager linkCheckManager = new LinkCheckManager();
-		linkCheckManager.setDbAdaptor(this.dbAdapter);
-		// Filter by references database name.
-		for (GKInstance refDBInst : LinksToCheckCache.getCache().keySet() )
+	private String checkLinksForRefDB(LinkCheckManager linkCheckManager, GKInstance refDBInst)
+	{
+		 int numLinkOK = 0;
+		 int numLinkNotOK = 0;
+		logger.info("Link-checking for database: {}", refDBInst.getDisplayName());
+		StringBuilder reportLine = new StringBuilder();
+		if (!refDBInst.getDisplayName().toUpperCase().contains(ENSEMBL))
+		{
+			reportLine.append(refDBInst.getDisplayName()).append("\t");
+		}
+		else
+		{
+			try
+			{
+				Optional<String> longerName = ((List<String>)refDBInst.getAttributeValuesList(ReactomeJavaConstants.name)).stream().filter(name -> name.length() > ENSEMBL.length()).sorted().findFirst();
+				reportLine.append(longerName.orElse(ENSEMBL)).append("\t");
+			}
+			catch (InvalidAttributeException e)
+			{
+				e.printStackTrace();
+			}
+			catch (Exception e)
+			{
+				e.printStackTrace();
+			}
+
+		}
+		Map<String, LinkCheckInfo> results = linkCheckManager.checkLinks(refDBInst, new ArrayList<>(LinksToCheckCache.removeRefDBFromCache(refDBInst)), this.proportionToLinkCheck, this.maxNumberLinksToCheck);
+		// "results" is a map of DB IDs mapped to link-checking results, for each identifier.
+		for (Entry<String, LinkCheckInfo> entry : results.entrySet())
 		{
-			StringBuilder reportLine = new StringBuilder();
-			int numLinkOK = 0;
-			int numLinkNotOK = 0;
-			// LinksToCheckCache.getRefDBsToCheck() should return a list that contains everything
-			// from the Spring file AND all of the ENSEMBL and KEGG species-specific reference database names.
-			if (LinksToCheckCache.getRefDBsToCheck().contains(refDBInst.getDisplayName())
-					|| (refDBInst.getDisplayName().toUpperCase().contains(ENSEMBL) && LinksToCheckCache.getRefDBsToCheck().contains(ENSEMBL))
-					|| (refDBInst.getDisplayName().toUpperCase().contains(KEGG) && LinksToCheckCache.getRefDBsToCheck().contains(KEGG))
-				)
+			LinkCheckInfo result = entry.getValue();
+			int statusCode = result.getStatusCode();
+			String identifier = result.getIdentifier();
+			// If the keyword is NOT found...
+			if (!result.isKeywordFound())
```

Comment:
In the future maybe a contingency on results that don't have a keyword could just be the presence of a '200' response? Resources like VGNC won't return a result with the keyword contained since it actually retrieves the data once you visit the URL. Would it make sense to have something along the lines of 'keyword was not found but a 200 status was received'? I know that given the way some bioinformatics resources are made, this will actually allow some incorrect ones to squeak through, but curious about if this is something we should consider

Reply:
_Just_ getting a 200 response code isn't terribly useful. There was one resource (won't name names but it rhymes with "leg" ;) ) that would give a 200 response with a message containing the identifier. And that message? It said the identifier could not be found.

I'd really prefer to define success a little more narrowly than just receiving a 200 response code.

We could revisit this in the future if it becomes an issue, but it's a bit beyond the scope of this PR.

Reply:
Yea fair enough, I saw after I'd written this comment how you whittle down the success to 'the response contains the identifier' && its '200 || 300ish' -- I just wanted to see what you had to sat about it